tests/test_pickle_core.py

```diff
@@ -33,7 +33,7 @@
 
 # Pickle core tests
 
-@cachier(next_time=False)
+@cachier(next_time=False, separate_files=True)
```

Comment:
Ok. You have to adjust all tests to run **both with `separate_files=True` and with `separate_files=False`**!

Read here on how to parameterize tests:
https://docs.pytest.org/en/6.2.x/parametrize.html

This might be a little more complicated here, because we want to paramtereize a decorator to a functionm but we have to find a way to do this without duplicating code.
If you get stuck with this please reach out to me for help.

Reply:
done.